src/Orchard.Web/Modules/Orchard.Widgets/Layouts/Providers/WidgetElementHarvester.cs

```diff
@@ -185,6 +186,18 @@ public class WidgetElementHarvester : Component, IElementHarvester {
                 return;
 
             var widget = context.Session.GetItemFromSession(widgetIdentity);
+
+            // I need to create a new widget and save the new one.
```

Comment:
Avoid using "I" in comments. Prefer the past form: "A new widget needs to be created and saved".
Then read all the comments you have ever written and see these "I"s in most of them ;)